dsl_parser/import_resolver/default_import_resolver.py

```diff
@@ -0,0 +1,142 @@
+
+    The resolver will go over the rules and for each matching rule
+    (its key is a prefix of the url) it will replace the prefix
+    with the value and will try to resolve the new url.
+
+    For example:
+        The rules list: [
+            {'http://prefix1': 'http://prefix1_replacement'},
+            {'http://prefix2': 'http://prefix2_replacement1'},
+            {'http://prefix2': 'http://prefix2_replacement2'}
+        ]
+        contains three rules that can be used for resolve URLs that
+        starts with 'http://prefix1' and 'http://prefix2'.
+        If the url is 'http://prefix2.suffix2.org' than the resolve method
+        will find a match in both the second and the third rules.
+
+        It will first try to apply the second rule by replacing the url's
+        prefix with the second rule value ('http://prefix2_replacement1')
+        and will try to resolve the new url:
+        'http://prefix2_replacement1.suffix2.org'.
+
+        In case this url cannot be resolved, it will try to apply
+        the third rule by replacing the url's prefix with
+        the third rule value ('http://prefix2_replacement2')
+        and will try to resolve the url:
+        'http://prefix2_replacement2.suffix2.org'.
+
+        If this url, also, cannot be resolved,
+        it will try to resolve the original url,
+        i.e. http://prefix2.suffix2.org'
+
+        In case that all the resolve attempts will fail,
+        a DSLParsingLogicException will be raise.
+    """
+
+    def __init__(self, rules=None):
+        # set the rules
+        self.rules = rules
+        if rules is None:
+            self.rules = DEFAULT_RULES
+        self._validate_rules()
+
+    def resolve(self, import_url):
+        failed_urls = {}
+        # trying to find a matching rule that can resolve this url
+        for rule in self.rules:
+            key = rule.keys()[0]
+            value = rule.values()[0]
+            prefix = key
+            prefix_len = len(key)
+            if prefix == import_url[:prefix_len]:
+                # found a matching rule
+                url_to_resolve = value + import_url[prefix_len:]
+                # trying to resolve the resolved_url
+                if url_to_resolve not in failed_urls.keys():
+                    # there is no point to try to resolve the same url twice
+                    try:
+                        return read_import(url_to_resolve)
+                    except DSLParsingLogicException, ex:
+                        # failed to resolve current rule,
+                        # continue to the next one
+                        failed_urls[url_to_resolve] = str(ex)
+
+        # failed to resolve the url using the rules
+        # trying to open the original url
+        try:
+            return read_import(import_url)
+        except DSLParsingLogicException, ex:
+            if not self.rules:
+                raise ex
```

Comment:
if you are simply re-raising, it is generally better to simply `raise` without any exception. This will re-`raise` the previous exception (which in this case would be the one currently named `ex`)